modules/library/src/main/scala/zio/elasticsearch/result/AggregationResult.scala

```diff
@@ -60,3 +60,21 @@ final case class TermsAggregationBucketResult private[elasticsearch] (
         Right(None)
     }
 }
+
+final case class FilterAggregationResult private[elasticsearch] (
+  docCount: Int,
+  subAggregations: Map[String, AggregationResult]
+) extends AggregationResult {
+
+  def subAggregationAs[A <: AggregationResult](aggName: String): Either[DecodingException, Option[A]] =
+    subAggregations.get(aggName) match {
+      case Some(aggRes) =>
+        Try(aggRes.asInstanceOf[A]) match {
+          case Failure(_)   => Left(DecodingException(s"Aggregation with name $aggName was not of type you provided."))
```

Comment:
This check will probably not work here because of type erasure: at runtime the type A is not known here, so `asInstanceOf` will allow anything and a cast error will most likely happen later in user code. An example to demonstrate the case: https://scastie.scala-lang.org/kyL2fVL2S8eQByJb4aS1Yg